infrastructure/src/main/java/org/corfudb/infrastructure/log/StreamLog.java

```diff
@@ -41,6 +42,15 @@
      */
     LogData read(long address);
 
+    /**
+     * Inspect if the stream log contains the entry at given address.
+     *
+     * @param address the address to inspect
+     * @return true if stream log contains the entry at the address, false otherwise
+     * @throws TrimmedException if address less than starting address
+     */
+    boolean contains(long address) throws TrimmedException;
```

Comment:
please don't throw an exception, just return an enum if you really need to know about trimm

Reply:
`contains` must not throw any exceptions, else it's is fragile and weird logic

Reply:
If you look at methods, like read/write, they throw exceptions like DataCorruptionException, OverwriteException, and catch them. They just don't mention it in the interface. It's actully the same. Here TrimmedException is important for upper layer to catch, so adding it here as a heads up.